src/OGL3X/UniformBlock.cpp

```diff
@@ -237,7 +258,17 @@ void UniformBlock::updateLightParameters()
 		m_currentBuffer = m_lightBlock.m_buffer;
 		glBindBuffer(GL_UNIFORM_BUFFER, m_lightBlock.m_buffer);
 	}
-	glBufferSubData(GL_UNIFORM_BUFFER, m_lightBlock.m_offsets[luLightDirection], m_lightBlockData.size(), pData);
+
+        OGLVideo & ogl = video();
+        OGLRender & render = ogl.getRender();
+        if (render.getRenderer() == OGLRender::glrAdreno)
+        {
+                glBufferData(GL_UNIFORM_BUFFER, m_lightBlockData.size(), m_lightBlockData.data(), GL_STATIC_COPY);
+        }
+        else
+        {
+                glBufferSubData(GL_UNIFORM_BUFFER, m_lightBlock.m_offsets[luLightDirection], m_lightBlockData.size(), pData);
```

Comment:
Same as above, this takes an offset, but we still copy all the data in m_lightBlockData? Shouldn't it be copying a subset of it?